src/overlays/actors/ovl_En_Minifrog/z_en_minifrog.c

```diff
@@ -21,60 +43,602 @@ const ActorInit En_Minifrog_InitVars = {
+void EnMinifrog_SpawnGrowAndShrink(EnMinifrog* this, GlobalContext* globalCtx) {
+    EnMinifrog_Jump(this);
+    if (this->timer > 0) {
+        Actor_SetScale(&this->actor, (0.01f + 0.0003f * this->timer * Math_SinS(0x1000 * (this->timer & 7))));
+        this->timer--;
+    } else {
+        Actor_SetScale(&this->actor, 0.01f);
+        this->actionFunc = EnMinifrog_Idle;
+    }
+}
+
+void EnMinifrog_Idle(EnMinifrog* this, GlobalContext* globalCtx) {
+    EnMinifrog_TurnToPlayer(this);
+    EnMinifrog_Jump(this);
+    EnMinifrog_JumpTimer(this);
+    if (func_800B84D0(&this->actor, globalCtx)) {
+        this->actionFunc = EnMinifrog_ReturnFrogCutscene;
+        if (this->actor.cutscene != -1) {
+            this->flags |= 1;
+        }
+    } else if ((this->actor.xzDistToPlayer < 100.0f) && (Actor_IsLinkFacingActor(&this->actor, 0x3000, globalCtx) != 0) && (func_8012403C(globalCtx) == 0xD)) {
+        func_800B8614(&this->actor, globalCtx, 110.0f);
+    }
+}
+
+void EnMinifrog_SetupNextFrogInit(EnMinifrog* this, GlobalContext* globalCtx) {
+    struct EnMinifrog* nextFrog;
+    struct EnMinifrog* missingFrog;
+
+    EnMinifrog_Jump(this);
+    nextFrog = this->frog;
+    if (nextFrog != NULL) {
+        missingFrog = nextFrog->frog;
+        if (nextFrog->frog != NULL) {
+            this->actor.home.rot.y = (s16)Actor_YawBetweenActors(&this->actor, &missingFrog->actor); // Set home to missing frog
+            EnMinifrog_TurnToMissingFrog(this);
+        } else {
+            EnMinifrog_TurnToPlayer(this);
+        }
+
+        if (this->frog->actor.home.rot.z == (this->actor.params & 0xF)) {
+            EnMinifrog_SetJumpState(this);
+            this->frog->actor.home.rot.z = 0;
+        }
+    }
+}
+
+void EnMinifrog_CheckChoirSuccess(EnMinifrog* this, GlobalContext* globalCtx) {
+    this->actionFunc = EnMinifrog_YellowFrogDialog;
+    if (this->frog != NULL) {
+        func_80151938(globalCtx, 0xD78); // "Unfortunately, it seems not all of our members have gathered."
+    } else {
+        func_80151938(globalCtx, 0xD7C); // "The conducting was spectacular. And all of our members rose to the occasion!"
+    }
+
+    EnMinifrog_SetJumpState(this);
+    this->frog = NULL;
+    this->actor.home.rot.z = 0;
+}
+
+// Needs returns to match
+void EnMinifrog_ContinueChoirCutscene(EnMinifrog* this, GlobalContext* globalCtx) {
+    EnMinifrog_Jump(this);
+    if (ActorCutscene_GetCurrentIndex() == 0x7C) {
+        EnMinifrog_CheckChoirSuccess(this, globalCtx);
+        return;
+    } else if (ActorCutscene_GetCanPlayNext(0x7C)) {
+        ActorCutscene_Start(0x7C, NULL);
+        EnMinifrog_CheckChoirSuccess(this, globalCtx);
+        return;
+    } else if (this->actor.cutscene != -1 && ActorCutscene_GetCurrentIndex() == this->actor.cutscene) {
+        ActorCutscene_Stop(this->actor.cutscene);
+        ActorCutscene_SetIntentToPlay(0x7C);
+        return;
+    } else {
+        ActorCutscene_SetIntentToPlay(0x7C);
+    }
```

Comment:
```suggestion
    if (ActorCutscene_GetCurrentIndex() == 0x7C) {
        EnMinifrog_CheckChoirSuccess(this, globalCtx);
    } else if (ActorCutscene_GetCanPlayNext(0x7C)) {
        ActorCutscene_Start(0x7C, NULL);
        EnMinifrog_CheckChoirSuccess(this, globalCtx);
    } else if (this->actor.cutscene != -1 && ActorCutscene_GetCurrentIndex() == this->actor.cutscene) {
        ActorCutscene_Stop(this->actor.cutscene);
        ActorCutscene_SetIntentToPlay(0x7C);
    } else {
        ActorCutscene_SetIntentToPlay(0x7C);
    }
```

Reply:
I mentioned in a comment at the start of the function that the returns are required to match and I can't figure out why or how to match without them. I'll move the comment to `necessary to match` in front of the returns